.github/workflows/build.yml

```diff
@@ -19,10 +23,12 @@ jobs:
         run: npm test
       - name: Lint
         run: npm run lint
-      - name: Deploy to GitHub Pages
-        uses: peaceiris/actions-gh-pages@v3
+      - name: Upload build to GitHub Pages
```

Comment:
Done.
You can see the CI of a commit on the forked branch [here](https://github.com/IEduStu/stremio-web/actions/runs/8760374269).
I've make this change to the CI to make it deploy the build to GitHub Pages, since the existing action for this didn't work.

This new code uses GitHub Action's official actions to deploy to GitHub Pages.
To make it work for any fork, you just have to enable GitHub Actions on the fork and enable GitHub Pages in the repo settings and it would work without configuring any environment variable or something like that.

Reply:
I originally made this change because I thought the deployment didn't work, but it seems that it did just in a different manner than I anticipated.
This is really cool :)

I see how useful it is to have it for a repo with multiple developers working on it, but I think it may lead other forks the same path that I went - thinking the deployment didn't work.
So I made another change for the CI to use the previous action when an environment variable named `GITHUB_PAGES_MULTI_BRANCH` is set to `true`, and otherwise use the simpler deploy action.

This way you can set `GITHUB_PAGES_MULTI_BRANCH` to `true` in this repo settings and preserve the current behavior, and forks that people create won't inherit this environment variable and thus will have the simpler deploy action.
The idea of this is to remove friction in developing this repo on forks, while preserving the current CI behavior.

Let me know if you think it's a good idea or whether should I just revert this change.